### PR TITLE
Number generated task-list items with an escaped dot

### PR DESCRIPTION
**Pull request by 0xGosu**
## Summary
- `RenderGeneratedTaskList` now prefixes each item in a per-agent `tasks.md` with a 1-based numbered ID and an escaped dot (`1\. `, `2\. `, …) instead of a plain bullet, e.g. `- [-] 1\. First task`, `- [ ] 2\. Second task`. The backslash keeps the marker from rendering as a Markdown ordered list and is the deliberate signal distinguishing a generator-written ID from operator-typed numbering in a hand-authored task-source file — this lets a standard markdown task-list parser using a digit/dot-hierarchy ID scheme (e.g. the task-list-md tool) read the file directly.
- `NextDeclaredTask`/`PendingDeclaredTasks` strip this exact marker back off when resolving the bare task text (regex `^\d+(?:\.\d+)*\\\.\s+`), so the daemon's declared-task flow keeps sending clean, unnumbered text to agents. Plain operator-typed numbering with no backslash (e.g. `- [ ] 1. do the thing`) is left untouched — existing behavior for hand-authored task sources is unaffected.
- Updated `internal/domain/taskgen_test.go` for the new render format and two `internal/frontend/frontend_test.go` tests that asserted the old literal (unnumbered) file content.

## Test plan
- [x] `gofmt -l .` clean
- [x] `go build -tags "vectors cpu" ./...`
- [x] `go vet -tags "vectors cpu" ./...`
- [x] `go test -tags "vectors cpu" ./... -count=1` — full suite passes (22 packages)
- [x] Regex verified against edge cases (multi-digit IDs, hierarchical IDs, non-backslash numbering, no-collision with generated task text)

https://claude.ai/code/session_016LEnXqxPFzYhbr4HFuhZKN

### CODE REVIEW

File: internal/domain/taskgen.go [hunk 1]
```diff
@@ -208,6 +217,8 @@ func RenderGeneratedTaskList(agentName string, tasks []string) string {
 		b.WriteString("- ")
 		b.WriteString(marker)
 		b.WriteString(" ")
+		b.WriteString(strconv.Itoa(i + 1))
+		b.WriteString(`\. `)
```
[the remark this answers is not in the document]
Comment (0xGosu) [reply internal/domain/taskgen.go]: Confirmed — verified empirically against the actual `task_list_md.py` regex (`^(\s*)-\s*(\[[ \-x\+\*]\])\s*(\d+(?:\.\d+)*)\.?\s*(.*)$`): the backslash was left in the captured description group instead of being consumed as the separator. Fixed in 44ee97b — `RenderGeneratedTaskList` now emits a plain dot (`1. `, `2. `, …), and I removed the strip-back-off logic from `NextDeclaredTask`/`PendingDeclaredTasks` since it's no longer needed: a numbered generated file is now indistinguishable from a hand-authored numbered checklist, whose numbering was already sent to the agent verbatim before this PR. Re-verified against the real parser regex in isolation, plus full local test suite (22 packages) green.
